src/Domain/Donation.php

```diff
@@ -303,16 +305,31 @@ class Donation extends SalesforceWriteProxy
      * @deprecated but retained for now as used in old test classes. Not recommend for continued use - either use
      * fromApiModel or create a new named constructor that takes required data for your use case.
      */
-    public static function emptyTestDonation(string $amount, PaymentMethodType $paymentMethodType = PaymentMethodType::Card): self
+    public static function emptyTestDonation(string $amount, PaymentMethodType $paymentMethodType = PaymentMethodType::Card, string $currencyCode = 'GBP'): self
     {
-        return new self($amount, 'GBP', $paymentMethodType);
+        return new self($amount, $currencyCode, $paymentMethodType);
     }
 
     private function __construct(string $amount, string $currencyCode, PaymentMethodType $paymentMethodType)
     {
         $this->fundingWithdrawals = new ArrayCollection();
-        $this->setCurrencyCode($currencyCode);
-        $this->setAmountAndMethod($amount, $paymentMethodType);
+        $this->currencyCode = $currencyCode;
+        $maximumAmount = self::maximumAmount($paymentMethodType);
+
+        if (
+            bccomp($amount, (string)$this->minimumAmount, 2) === -1 ||
+            bccomp($amount, (string)$maximumAmount, 2) === 1
+        ) {
+            throw new \UnexpectedValueException(sprintf(
+                'Amount must be %d-%d %s',
+                $this->minimumAmount,
+                $maximumAmount,
+                $this->currencyCode,
+            ));
+        }
+
+        $this->amount = $amount;
+        $this->paymentMethodType = $paymentMethodType;
```

Comment:
Psalm only allows assignment to readonly properties from inside the constructor, hence inlining the `setAmountAndMethod` function.